### ISSUE TITLE
joging GRBL with a rotary dial (MPG)

### ISSUE THREAD

**Issue by samitray** [grbl/grbl#243]
This is not an issue, hence may be posted in the wrong list.

Have a  MPG pendant with axis and speed selector lying around.
would like to use it with my set-up Rpi+UGS+GRBL on Uno

I am thinking about adding another uno to read pulses from dial, axis selector
and and send commands to grbl similar to keypad jog in UGS. 
Sending commands on the same channel is the challenge since UGS is already
connected to GRBL - Uno serial port.
not able to figure how to get the communication sorted.
has anybody figured out jogging grbl with a dial?

**Comment by yaplej** [comment]
@sacidu93 Great job on that.  It looks amazing.

**Comment by stealth027** [comment]
Hi All, 

This is all amazing, it really is, but I am just starting out and have no idea what everyone is actually referring to... I am currently trying to connect a nano to my grbl, just to build a simple controller with lcd readout on position. Could someone please point me in the right direction. I know that between the lot of you, you could do this with your eyes closed, buy I have no idea. I would seriously appreciate all the help you could give me.

**Comment by TadyTheFish** [comment]
I'm currently building and testing my own interface. I too need an MPG pendant and i know how to program in the arduino enviroment. Sacidu93 posted pictures of his and i really like it. But there is a big problem when connecting multiple devices in parallel on the serial bus. I really love using the grblControl program for sending g-code but even in idle the serial bus is busy with traffic of constant updates od position.So i used a different aproach. I used a Mega2560 for my pendant. And the PC connects to it. So the data passes troug the pendant and out the second serial post of the mega ( it has multiple hardware serial ports). I was suprised at the results. I ran a few g-codes without error!. So now i made a program that reads an MPG and serial prints to the second port (it stops com with computer). It sends 
G0G91(axys)(value)
G90
?

The thing works very good. I was scared that the data would have a delay or get corrupted. But it wirk:)
I really hope the new version of Grbl will suport MPGs;)
